public/API/API_GetGameFromHash.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ *  API_GetGameFromHash - looks up a game from a given hash string
+ *    h : game hash
+ *
+ *  int        ID                         id of the game
+ *  string     Title                      name of the game
+ *  int        ConsoleID                  unique identifier of the console associated to the game
+ *  string     ConsoleName                name of the console associated to the game
+ */
+
+use App\Platform\Models\GameHash;
+use App\Platform\Models\System;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Validator;
+
+$input = Validator::validate(Arr::wrap(request()->query()), [
+    'h' => ['required', 'size:32'],
+]);
+
+$givenHash = request()->query('h');
+
+$foundHash = GameHash::with('game')->where('MD5', $givenHash)->first();
+
+if (!$foundHash) {
+    return response()->json(['message' => 'Could not find a game matching the given hash.'], 400);
```

Comment:
Recommend using `error` instead of `message`, so it's at least consistent with the API_GetTicketData endpoint:
https://github.com/RetroAchievements/RAWeb/blob/3d5d845ea70737f90d9a35eb867111a87018d387/public/API/API_GetTicketData.php#L154

API_GetAchievementUnlocks, API_GetUserPoints, and API_GetUserSummary just echo back the inputs on a 404 not found response.

Also consider a more expanded response as described [here](https://discord.com/channels/476211979464343552/757767535293890682/1132370669938671758).

```
HTTP/1.1 404 Not Found
{
  "status": 404,
  "code": "not_found",
  "error": "Unknown hash: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
}
```
> This kinda matches or at least come closest to a JSON:API style response (https://jsonapi.org/format/#error-objects); plans are to do a similar thing for the Web API.

@luchaos probably has a specific structure in mind.

Reply:
Switched to `error` and the expanded response structure in https://github.com/RetroAchievements/RAWeb/pull/2090/commits/8ec90f1122c9f8f7847bdfdebacc32a8be1e9d4a.